core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -920,6 +927,11 @@ private void acquirePrimaryShardReference(ShardId shardId, String allocationId,
         if (actualAllocationId.equals(allocationId) == false) {
             throw new ShardNotFoundException(shardId, "expected aID [{}] but found [{}]", allocationId, actualAllocationId);
         }
+        final long actualTerm = indexShard.getPrimaryTerm();
```

Comment:
why did you choose to do it here rather than doing it in index shard, like we discussed?